lacework/resource_lacework_integration_aws_eks_audit_log.go

```diff
@@ -0,0 +1,237 @@
+package lacework
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/lacework/go-sdk/api"
+)
+
+func resourceLaceworkIntegrationAwsEksAuditLog() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceLaceworkIntegrationAwsEksAuditLogCreate,
+		Read:     resourceLaceworkIntegrationAwsEksAuditLogRead,
+		Update:   resourceLaceworkIntegrationAwsEksAuditLogUpdate,
+		Delete:   resourceLaceworkIntegrationAwsEksAuditLogDelete,
+		Schema:   awsEksAuditLogIntegrationSchema,
+		Importer: &schema.ResourceImporter{State: importLaceworkIntegration},
+	}
+}
+
+var awsEksAuditLogIntegrationSchema = map[string]*schema.Schema{
+	"name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The integration name.",
+	},
+	"type_name": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"enabled": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     true,
+		Description: "The state of the external integration.",
+	},
+	"retries": {
+		Type:        schema.TypeInt,
+		Optional:    true,
+		Default:     5,
+		Description: "The number of attempts to create the external integration.",
+	},
+	"intg_guid": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"sns_arn": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The SNS ARN.",
+	},
+	"credentials": {
+		Type:        schema.TypeList,
+		MaxItems:    1,
+		Required:    true,
+		Description: "The credentials needed by the integration.",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"role_arn": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"external_id": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+			},
+		},
+	},
+	"is_org": {
+		Type:     schema.TypeBool,
+		Computed: true,
+	},
+	"created_or_updated_time": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"created_or_updated_by": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+}
+
+func resourceLaceworkIntegrationAwsEksAuditLogCreate(d *schema.ResourceData, meta interface{}) error {
+	var (
+		lacework           = meta.(*api.Client)
+		retries            = d.Get("retries").(int)
+		awsEksAuditLogData = api.AwsEksAuditData{
+			SnsArn: d.Get("snsArn").(string),
```

Comment:
Create does not work
panic: interface conversion: interface {} is nil, not string